recovery.md

```diff
@@ -0,0 +1,82 @@
+# The recovery partition
+
+On Ubuntu Core 20 systems there is a new system-recovery partition. It
+is big enough to contain a certain number of recovery "systems" that
+can then be used to recover or reinstall a broken system.
+
+A full "seed" to reinstall on a generic amd64 system is roughly 280MB
+so the recovery partition is sized to 600MB to be able to store two
+recovery systems by default.
+
+The system-recovery partition is next to the system-boot
+partition. Both are vfat as required by UEFI.
+
+On full disk encrypted systems the system-boot partition contains the
+unpacked kernel(s) to boot init initramfs that then unencrypts the
+"writable" partition. The partitions are separate because we want to
+write to the recovery partition as rarely as possible to avoid
+filesystem issues.
+
+# Recovery system file layout
+
+The revery partition contains the following file layout:
+
+/system/<name>/snaps/{base.snap,kernel.snap,other...}
```

Comment:
<name> should include the branch/model (directories) - we need to consider upper/lower. What to boot initially needs to be written by `snap prepare-image` in grubenv

Reply:
I can double check what are the relevant regexps in snapd

Reply:
* snap names should be like  `^[a-z0-9-]*[a-z][a-z0-9-]*$`  and at most 40 chars
* model names should be like `^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$` but we enforce lowercase on top of that, no size limits ATM afaict
* brand names  should be like `^(?:[a-z0-9A-Z]{32}|[-a-z0-9]{2,28})$`,  there sadly  uppercase can happen, same problem with snap-ids

Reply:
Uh, so in theory we might have "Brand" and "brand" and on vfat its the same - that is very unfortunate, I think that means we need to reconsider using branch/model in the directory.

Reply:
yes, but the example you gave is misleading,  nice brand names are enforced lowercase (like "canonical"), it's the random ones (same as snap ids) that can have uppercase, we can't assume the brand will be nice, is things like CA5GLZgNQWPhspDQK63Er46Uxz2SO7ez that have uppercase in them, it's unlikely they will collide but not impossible